### PR TITLE
[BugFix][Sleep] Keep a dedicated 1 MiB HCCL lifecycle anchor

### PR DESCRIPTION
**Pull request by yjyang62**
### What this PR does / why we need it?

`rl_config.sleep_mode_extra_cleanup` destroys HCCL process groups during sleep so EP/MC2 windows can be returned to the trainer. Destroying every multi-rank communicator can tear down shared CANN HCCP/AICPU state. After restore, the first DSA metadata op (`SparseAttnSharedkvMetadata`) then fails and `wake_up` returns 500.

This PR does not keep a business group (TP/DP/EP/world). Before teardown it creates a dedicated world-size HCCL group named `sleep_lifecycle_anchor` with a **1 MiB** buffer, initializes it with a device `barrier`, and uses that group as a temporary lifecycle nail:

1. Sleep creates the coordinator on the first cycle, or `restore_hccl()` on later cycles if the communicator was released after the previous wake-up.
2. A device barrier physically initializes the nail (HCCL `new_group` can be lazy).
3. All other HCCL groups are destroyed. If the nail cannot be established, teardown is skipped for that cycle.
4. Wake-up restores business groups and refreshes MoE/MC2 metadata. The nail stays up through ACL Graph recapture.
5. After recapture (or after the `kv_cache` stage of level-2 wake-up), all ranks barrier on the nail and `destroy_hccl()` it. The Python coordinator is kept (`device_group=None`) and reused next sleep, so the same group name is not registered twice.

`get_hccl_config_for_pg_options` returns the 1 MiB size for `sleep_lifecycle_anchor` **before** building the normal config map, so sleep-time creation does not call `calculate_dp_buffer_size()` / `get_current_vllm_config()`.

### Does this PR introduce _any_ user-facing change?

Yes. Extra-cleanup creates one temporary 1 MiB world-size HCCL group for each sleep/wake cycle and releases it after graph recapture. There is no new environment variable or `rl_config` key.

### How was this patch tested?

- Unit tests in `tests/ut/device_allocator/test_sleep_mem_optimized.py`:
  - first-cycle create + reuse / restore of the existing coordinator
  - destroy/restore business groups while skipping the nail
  - skip all HCCL teardown when nail init fails
  - release after recapture; keep the nail across staged `tags=["weights"]` wake-up

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

**Comment by yjyang62** [comment]
/rerun
[Bot]: rerun completed.

**Rerun (failed jobs only):**
- E2E

File: vllm_ascend/device_allocator/sleep_mem_optimized.py [hunk 1]
```diff
@@ -143,18 +151,90 @@ def iter_alive_group_coordinators():
...
+                ranks = list(range(world_size))
+                self._lifecycle_anchor_group = init_model_parallel_group(
+                    [ranks],
+                    get_world_group().local_rank,
+                    "hccl",
+                    use_device_communicator=False,
+                    group_name=SLEEP_LIFECYCLE_ANCHOR_GROUP_NAME,
+                )
+
+            device_group = getattr(self._lifecycle_anchor_group, "device_group", None)
```
[the remark this answers is not in the document]
Comment (yjyang62) [reply vllm_ascend/device_allocator/sleep_mem_optimized.py:179]: Thanks. `init_model_parallel_group` in vLLM returns a `GroupCoordinator` or raises; it is not expected to return `None`.

Even if it did, `_ensure_lifecycle_anchor` is already wrapped in `except Exception`: an `AttributeError` on `restore_hccl()` is logged and the function returns `False`. `destroy_hccl()` then sets `_skip_hccl_cleanup_for_cycle` and skips tearing down business groups, so this is not an unhandled crash on the sleep path.

File: vllm_ascend/device_allocator/sleep_mem_optimized.py [hunk 2]
```diff
@@ -24,11 +24,12 @@
 
 import torch
 from vllm.config import VllmConfig, set_current_vllm_config
-from vllm.distributed.parallel_state import _groups
+from vllm.distributed.parallel_state import _groups, get_world_group, init_model_parallel_group
 from vllm.logger import logger
 from vllm.utils.mem_constants import GiB_bytes
 
 from vllm_ascend.compilation import acl_graph
+from vllm_ascend.utils import SLEEP_LIFECYCLE_ANCHOR_GROUP_NAME
```
[the remark this answers is not in the document]
Comment (yjyang62) [reply vllm_ascend/device_allocator/sleep_mem_optimized.py:32]: `[BugFix][Sleep] ...` matches the project convention in `AGENTS.md` (`[Type][Module] Description`). The base branch is main, so the Branch prefix is omitted. No title change needed.

**Comment by yjyang62** [comment]
Destroying every HCCL group during sleep extra-cleanup tears down HCCP. Wake brings HCCP back, and the new HCCP process overwrites the topic PID slot that custom AICPU uses. The next custom metadata task (SparseAttnSharedkvMetadata) is delivered to HCCP. The driver rejects it because the group schedule mode does not match, and Runtime reports AICPU 507018. This is a dispatch failure, not a bad metadata tensor.

Measured on device 0, host PID 3730078, same table_id=1, same pid_type=3:

custom AICPU (cp_type=1, device PID 21702) registers the slot
restarted HCCP (cp_type=4, device PID 21800) writes the same slot
The PIDs are different. They share one topic-table entry. The following dispatch log is pid=21800; gid=0; grp_ctx_sched_mode=0; sched_mode=1. esched_drv_get_grp requires SCHED_MODE_SCHED_CPU (1). HCCP group 0 is still uninitialized (0), so the driver returns NULL and esched_aicpu_sched_task fails to publish the event. The same chain shows up on the full service: TS dump has device pid user equal to the new HCCP PID, with the same mode mismatch.

The public driver mapping that produces this is in esched_drv_config_pid: DEVDRV_PROCESS_CP1 and DEVDRV_PROCESS_QS go to DEVICE_STD_PROC (pid_type=2); every other device process, including custom AICPU (CP2, cp_type=1) and HCCP (cp_type=4), goes to DEVICE_USER_PROC (pid_type=3). topic_sched_config_pid then looks up one row by host_pid and stores that value in STARS_TOPIC_DEVICE_PID_USER. Last writer wins. 

Two other paths keep working, which is why the first overwrite does not fail:

HCCP control traffic does not reuse this topic slot. RaHdcInitSessionConnectEx queries the devdrv process table by DEVDRV_PROCESS_HCCP (cp_type=4), then uses the existing HDC session. Custom AICPU and HCCP are separate entries there.
This run's HCCL AllReduce runs on the normal AICPU scheduler (pid_type=2, isCustom=0), not on the custom slot.
Order explains the asymmetry. First init registers HCCP, then custom AICPU, so the slot ends on custom AICPU. Sleep keeps custom AICPU alive and restarts HCCP, so the slot ends on HCCP.

The anchor in this PR does not change that mapping. It keeps one initialized world-size HCCL group across teardown so HCCP never exits, and the custom slot is not overwritten. A minimal repro (no model, no CaMem, no graph recapture) fails exactly after the new group performs a real allreduce, and passes when another live communicator is retained.

**Review by Ronald1995 — APPROVED** [review]
LGTM

**Review by ZT-AIA — APPROVED** [review]
